GridView makes application crash

I'm using controlsfx v11.1.0 inside a simple javafx project. I add a GridView to my fxml file and reference it in  my view's controller. But when I try to set a observable list to it, my app crashes with a cryptic error message that I get 0 google results for. 

```
Exception in thread "main" java.lang.RuntimeException: Exception in Application start method
	at com.sun.javafx.application.LauncherImpl.launchApplication1(LauncherImpl.java:901)
	at com.sun.javafx.application.LauncherImpl.lambda$launchApplication$2(LauncherImpl.java:196)
	at java.base/java.lang.Thread.run(Thread.java:831)
Caused by: java.lang.NullPointerException: Cannot invoke "javafx.scene.control.IndexedCell.updateIndex(int)" because "<parameter1>" is null
	at javafx.scene.control.skin.VirtualFlow.setCellIndex(VirtualFlow.java:1807)
	at javafx.scene.control.skin.VirtualFlow.getCell(VirtualFlow.java:1791)
	at javafx.scene.control.skin.VirtualFlow.getOrCreateCellSize(VirtualFlow.java:2966)
	at javafx.scene.control.skin.VirtualFlow.getOrCreateCellSize(VirtualFlow.java:2949)
	at javafx.scene.control.skin.VirtualFlow.recalculateAndImproveEstimatedSize(VirtualFlow.java:3021)
	at javafx.scene.control.skin.VirtualFlow.recalculateEstimatedSize(VirtualFlow.java:3013)
	at javafx.scene.control.skin.VirtualFlow.layoutChildren(VirtualFlow.java:1052)
	at javafx.scene.control.skin.VirtualFlow$5.invalidated(VirtualFlow.java:885)
	at javafx.beans.property.IntegerPropertyBase.markInvalid(IntegerPropertyBase.java:113)
	at javafx.beans.property.IntegerPropertyBase.set(IntegerPropertyBase.java:148)
	at javafx.scene.control.skin.VirtualFlow.setCellCount(VirtualFlow.java:899)
	at impl.org.controlsfx.skin.GridViewSkin.updateItemCount(GridViewSkin.java:134)
	at impl.org.controlsfx.skin.GridViewSkin.updateGridViewItems(GridViewSkin.java:99)
	at impl.org.controlsfx.skin.GridViewSkin.<init>(GridViewSkin.java:52)
	at org.controlsfx.control.GridView.createDefaultSkin(GridView.java:142)
	at javafx.scene.control.Control.doProcessCSS(Control.java:897)
	at javafx.scene.control.Control$1.doProcessCSS(Control.java:89)
	at com.sun.javafx.scene.control.ControlHelper.processCSSImpl(ControlHelper.java:67)
	at com.sun.javafx.scene.NodeHelper.processCSS(NodeHelper.java:146)
	at javafx.scene.Parent.doProcessCSS(Parent.java:1400)
	at javafx.scene.Parent$1.doProcessCSS(Parent.java:125)
	at com.sun.javafx.scene.ParentHelper.processCSSImpl(ParentHelper.java:98)
	at com.sun.javafx.scene.NodeHelper.processCSS(NodeHelper.java:146)
	at javafx.scene.Parent.doProcessCSS(Parent.java:1400)
	at javafx.scene.Parent$1.doProcessCSS(Parent.java:125)
	at com.sun.javafx.scene.ParentHelper.processCSSImpl(ParentHelper.java:98)
	at com.sun.javafx.scene.NodeHelper.processCSS(NodeHelper.java:146)
	at javafx.scene.Parent.doProcessCSS(Parent.java:1400)
	at javafx.scene.Parent$1.doProcessCSS(Parent.java:125)
	at com.sun.javafx.scene.ParentHelper.processCSSImpl(ParentHelper.java:98)
	at com.sun.javafx.scene.NodeHelper.processCSS(NodeHelper.java:146)
	at javafx.scene.Parent.doProcessCSS(Parent.java:1400)
	at javafx.scene.Parent$1.doProcessCSS(Parent.java:125)
	at com.sun.javafx.scene.ParentHelper.processCSSImpl(ParentHelper.java:98)
	at com.sun.javafx.scene.NodeHelper.processCSS(NodeHelper.java:146)
	at javafx.scene.Parent.doProcessCSS(Parent.java:1400)
	at javafx.scene.Parent$1.doProcessCSS(Parent.java:125)
	at com.sun.javafx.scene.ParentHelper.processCSSImpl(ParentHelper.java:98)
	at com.sun.javafx.scene.control.ControlHelper.superProcessCSSImpl(ControlHelper.java:63)
	at com.sun.javafx.scene.control.ControlHelper.superProcessCSS(ControlHelper.java:55)
	at javafx.scene.control.Control.doProcessCSS(Control.java:900)
	at javafx.scene.control.Control$1.doProcessCSS(Control.java:89)
	at com.sun.javafx.scene.control.ControlHelper.processCSSImpl(ControlHelper.java:67)
	at com.sun.javafx.scene.NodeHelper.processCSS(NodeHelper.java:146)
	at javafx.scene.Parent.doProcessCSS(Parent.java:1400)
	at javafx.scene.Parent$1.doProcessCSS(Parent.java:125)
	at com.sun.javafx.scene.ParentHelper.processCSSImpl(ParentHelper.java:98)
	at com.sun.javafx.scene.NodeHelper.processCSS(NodeHelper.java:146)
	at javafx.scene.Node.processCSS(Node.java:9456)
	at javafx.scene.Scene.doCSSPass(Scene.java:572)
	at javafx.scene.Scene.preferredSize(Scene.java:1774)
	at javafx.scene.Scene$2.preferredSize(Scene.java:396)
	at com.sun.javafx.scene.SceneHelper.preferredSize(SceneHelper.java:66)
	at javafx.stage.Window$12.invalidated(Window.java:1114)
	at javafx.beans.property.BooleanPropertyBase.markInvalid(BooleanPropertyBase.java:110)
	at javafx.beans.property.BooleanPropertyBase.set(BooleanPropertyBase.java:145)
	at javafx.stage.Window.setShowing(Window.java:1190)
	at javafx.stage.Window.show(Window.java:1205)
	at javafx.stage.Stage.show(Stage.java:273)
	at com.example.MyApplication.start(MyApplication.kt:32)
	at com.sun.javafx.application.LauncherImpl.lambda$launchApplication1$9(LauncherImpl.java:847)
	at com.sun.javafx.application.PlatformImpl.lambda$runAndWait$12(PlatformImpl.java:484)
	at com.sun.javafx.application.PlatformImpl.lambda$runLater$10(PlatformImpl.java:457)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:391)
	at com.sun.javafx.application.PlatformImpl.lambda$runLater$11(PlatformImpl.java:456)
	at com.sun.glass.ui.InvokeLaterDispatcher$Future.run$$$capture(InvokeLaterDispatcher.java:96)
	at com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java)
	at com.sun.glass.ui.win.WinApplication._runLoop(Native Method)
	at com.sun.glass.ui.win.WinApplication.lambda$runLoop$3(WinApplication.java:184)
	... 1 more
```

I can confirm the bug for javafx 17.0.1 and the workaround using Platform.runLater works.
```

Caused by: java.lang.NullPointerException: Cannot invoke "javafx.scene.control.IndexedCell.updateIndex(int)" because "<parameter1>" is null
	at javafx.scene.control.skin.VirtualFlow.setCellIndex(VirtualFlow.java:1807) ~[javafx-controls-17.0.1-mac.jar:?]
	at javafx.scene.control.skin.VirtualFlow.getCell(VirtualFlow.java:1791) ~[javafx-controls-17.0.1-mac.jar:?]
	at javafx.scene.control.skin.VirtualFlow.getOrCreateCellSize(VirtualFlow.java:2966) ~[javafx-controls-17.0.1-mac.jar:?]
	at javafx.scene.control.skin.VirtualFlow.getOrCreateCellSize(VirtualFlow.java:2949) ~[javafx-controls-17.0.1-mac.jar:?]
	at javafx.scene.control.skin.VirtualFlow.recalculateAndImproveEstimatedSize(VirtualFlow.java:3021) ~[javafx-controls-17.0.1-mac.jar:?]
	at javafx.scene.control.skin.VirtualFlow.recalculateEstimatedSize(VirtualFlow.java:3013) ~[javafx-controls-17.0.1-mac.jar:?]
	at javafx.scene.control.skin.VirtualFlow.layoutChildren(VirtualFlow.java:1052) ~[javafx-controls-17.0.1-mac.jar:?]
	at javafx.scene.control.skin.VirtualFlow$5.invalidated(VirtualFlow.java:885) ~[javafx-controls-17.0.1-mac.jar:?]
	at javafx.beans.property.IntegerPropertyBase.markInvalid(IntegerPropertyBase.java:113) ~[javafx-base-17.0.1-mac.jar:?]
	at javafx.beans.property.IntegerPropertyBase.set(IntegerPropertyBase.java:148) ~[javafx-base-17.0.1-mac.jar:?]
	at javafx.scene.control.skin.VirtualFlow.setCellCount(VirtualFlow.java:899) ~[javafx-controls-17.0.1-mac.jar:?]
	at impl.org.controlsfx.skin.GridViewSkin.updateItemCount(GridViewSkin.java:134) ~[controlsfx-11.1.0.jar:?]
	at impl.org.controlsfx.skin.GridViewSkin.updateGridViewItems(GridViewSkin.java:99) ~[controlsfx-11.1.0.jar:?]
	at impl.org.controlsfx.skin.GridViewSkin.<init>(GridViewSkin.java:52) ~[controlsfx-11.1.0.jar:?]
	at org.controlsfx.control.GridView.createDefaultSkin(GridView.java:142) ~[controlsfx-11.1.0.jar:?]
	at javafx.scene.control.Control.doProcessCSS(Control.java:897) ~[javafx-controls-17.0.1-mac.jar:?]
	at javafx.scene.control.Control$1.doProcessCSS(Control.java:89) ~[javafx-controls-17.0.1-mac.jar:?]
	at com.sun.javafx.scene.control.ControlHelper.processCSSImpl(ControlHelper.java:67) ~[javafx-controls-17.0.1-mac.jar:?]
	at com.sun.javafx.scene.NodeHelper.processCSS(NodeHelper.java:146) ~[javafx-graphics-17.0.1-mac.jar:?]
	at javafx.scene.Parent.doProcessCSS(Parent.java:1400) ~[javafx-graphics-17.0.1-mac.jar:?]
	at javafx.scene.Parent$1.doProcessCSS(Parent.java:125) ~[javafx-graphics-17.0.1-mac.jar:?]
	at com.sun.javafx.scene.ParentHelper.processCSSImpl(ParentHelper.java:98) ~[javafx-graphics-17.0.1-mac.jar:?]
	at com.sun.javafx.scene.NodeHelper.processCSS(NodeHelper.java:146) ~[javafx-graphics-17.0.1-mac.jar:?]
	at javafx.scene.Parent.doProcessCSS(Parent.java:1400) ~[javafx-graphics-17.0.1-mac.jar:?]
	at javafx.scene.Parent$1.doProcessCSS(Parent.java:125) ~[javafx-graphics-17.0.1-mac.jar:?]
	at com.sun.javafx.scene.ParentHelper.processCSSImpl(ParentHelper.java:98) ~[javafx-graphics-17.0.1-mac.jar:?]
	at com.sun.javafx.scene.NodeHelper.processCSS(NodeHelper.java:146) ~[javafx-graphics-17.0.1-mac.jar:?]
	at javafx.scene.Parent.doProcessCSS(Parent.java:1400) ~[javafx-graphics-17.0.1-mac.jar:?]
	at javafx.scene.Parent$1.doProcessCSS(Parent.java:125) ~[javafx-graphics-17.0.1-mac.jar:?]
	at com.sun.javafx.scene.ParentHelper.processCSSImpl(ParentHelper.java:98) ~[javafx-graphics-17.0.1-mac.jar:?]
	at com.sun.javafx.scene.NodeHelper.processCSS(NodeHelper.java:146) ~[javafx-graphics-17.0.1-mac.jar:?]
	at javafx.scene.Parent.doProcessCSS(Parent.java:1400) ~[javafx-graphics-17.0.1-mac.jar:?]
	at javafx.scene.Parent$1.doProcessCSS(Parent.java:125) ~[javafx-graphics-17.0.1-mac.jar:?]
	at com.sun.javafx.scene.ParentHelper.processCSSImpl(ParentHelper.java:98) ~[javafx-graphics-17.0.1-mac.jar:?]
	at javafx.scene.control.PopupControl$CSSBridgeHelper.superProcessCSSImpl(PopupControl.java:1232) ~[javafx-controls-17.0.1-mac.jar:?]
	at javafx.scene.control.PopupControl$CSSBridgeHelper.superProcessCSS(PopupControl.java:1228) ~[javafx-controls-17.0.1-mac.jar:?]
	at javafx.scene.control.PopupControl$CSSBridge.doProcessCSS(PopupControl.java:1193) ~[javafx-controls-17.0.1-mac.jar:?]
	at javafx.scene.control.PopupControl$CSSBridgeHelper.processCSSImpl(PopupControl.java:1237) ~[javafx-controls-17.0.1-mac.jar:?]
	at com.sun.javafx.scene.NodeHelper.processCSS(NodeHelper.java:146) ~[javafx-graphics-17.0.1-mac.jar:?]
	at javafx.scene.Parent.doProcessCSS(Parent.java:1400) ~[javafx-graphics-17.0.1-mac.jar:?]
	at javafx.scene.Parent$1.doProcessCSS(Parent.java:125) ~[javafx-graphics-17.0.1-mac.jar:?]
	at com.sun.javafx.scene.ParentHelper.processCSSImpl(ParentHelper.java:98) ~[javafx-graphics-17.0.1-mac.jar:?]
	at com.sun.javafx.scene.NodeHelper.processCSS(NodeHelper.java:146) ~[javafx-graphics-17.0.1-mac.jar:?]
	at javafx.scene.Node.processCSS(Node.java:9456) ~[javafx-graphics-17.0.1-mac.jar:?]
	at javafx.scene.Scene.doCSSPass(Scene.java:572) ~[javafx-graphics-17.0.1-mac.jar:?]
	at javafx.scene.Scene.preferredSize(Scene.java:1774) ~[javafx-graphics-17.0.1-mac.jar:?]
	at javafx.scene.Scene$2.preferredSize(Scene.java:396) ~[javafx-graphics-17.0.1-mac.jar:?]
	at com.sun.javafx.scene.SceneHelper.preferredSize(SceneHelper.java:66) ~[javafx-graphics-17.0.1-mac.jar:?]
	at javafx.stage.Window$12.invalidated(Window.java:1114) ~[javafx-graphics-17.0.1-mac.jar:?]
	at javafx.beans.property.BooleanPropertyBase.markInvalid(BooleanPropertyBase.java:110) ~[javafx-base-17.0.1-mac.jar:?]
	at javafx.beans.property.BooleanPropertyBase.set(BooleanPropertyBase.java:145) ~[javafx-base-17.0.1-mac.jar:?]
	at javafx.stage.Window.setShowing(Window.java:1190) ~[javafx-graphics-17.0.1-mac.jar:?]
	at javafx.stage.Window.show(Window.java:1205) ~[javafx-graphics-17.0.1-mac.jar:?]
	at javafx.stage.PopupWindow.showImpl(PopupWindow.java:475) ~[javafx-graphics-17.0.1-mac.jar:?]
	at javafx.stage.PopupWindow.show(PopupWindow.java:420) ~[javafx-graphics-17.0.1-mac.jar:?]
	at org.controlsfx.control.PopOver.show(PopOver.java:463) ~[controlsfx-11.1.0.jar:?]
	at org.controlsfx.control.PopOver.show(PopOver.java:386) ~[controlsfx-11.1.0.jar:?]
	at org.controlsfx.control.PopOver.show(PopOver.java:320) ~[controlsfx-11.1.0.jar:?]
	at org.controlsfx.control.PopOver.show(PopOver.java:288) ~[controlsfx-11.1.0.jar:?]
	at org.jabref.gui.groups.GroupDialogView.openIconPicker(GroupDialogView.java:249) ~[main_/:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:567) ~[?:?]
	at com.sun.javafx.reflect.Trampoline.invoke(MethodUtil.java:77) ~[javafx-base-17.0.1-mac.jar:?]
	at jdk.internal.reflect.GeneratedMethodAccessor2.invoke(Unknown Source) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:567) ~[?:?]
	at com.sun.javafx.reflect.MethodUtil.invoke(MethodUtil.java:275) ~[javafx-base-17.0.1-mac.jar:?]
	at com.sun.javafx.fxml.MethodHelper.invoke(MethodHelper.java:84) ~[javafx-fxml-17.0.1-mac.jar:?]
	at javafx.fxml.FXMLLoader$MethodHandler.invoke(FXMLLoader.java:1854) ~[javafx-fxml-17.0.1-mac.jar:?]
```

--- a/controlsfx/src/main/java/impl/org/controlsfx/skin/GridRowSkin.java
+++ b/controlsfx/src/main/java/impl/org/controlsfx/skin/GridRowSkin.java
@@ -1,5 +1,5 @@
 /**
- * Copyright (c) 2013, 2018 ControlsFX
+ * Copyright (c) 2013, 2022, ControlsFX
  * All rights reserved.
  *
  * Redistribution and use in source and binary forms, with or without
@@ -68,7 +68,9 @@ public class GridRowSkin<T> extends CellSkinBase<GridRow<T>> {
         int rowIndex = getSkinnable().getIndex();
         if (rowIndex >= 0) {
             GridView<T> gridView = getSkinnable().getGridView();
-            int maxCellsInRow = ((GridViewSkin<?>)gridView.getSkin()).computeMaxCellsInRow();
+            GridViewSkin<?> gridViewSkin = (GridViewSkin<?>) gridView.getSkin();
+            if (gridViewSkin == null) return;
+            int maxCellsInRow = gridViewSkin.computeMaxCellsInRow();
             int totalCellsInGrid = gridView.getItems().size();
             int startCellIndex = rowIndex * maxCellsInRow;
             int endCellIndex = startCellIndex + maxCellsInRow - 1;

--- a/controlsfx/src/main/java/impl/org/controlsfx/skin/GridViewSkin.java
+++ b/controlsfx/src/main/java/impl/org/controlsfx/skin/GridViewSkin.java
@@ -1,5 +1,5 @@
 /**
- * Copyright (c) 2013, 2021, ControlsFX
+ * Copyright (c) 2013, 2022, ControlsFX
  * All rights reserved.
  *
  * Redistribution and use in source and binary forms, with or without
@@ -49,8 +49,6 @@ public class GridViewSkin<T> extends VirtualContainerBase<GridView<T>, GridRow<T
         super(control);
 
         flow = getVirtualFlow();
-        updateGridViewItems();
-
         flow.setId("virtual-flow"); //$NON-NLS-1$
         flow.setPannable(false);
         flow.setVertical(true);
@@ -58,6 +56,7 @@ public class GridViewSkin<T> extends VirtualContainerBase<GridView<T>, GridRow<T
         flow.setCellFactory(param -> createCell());
         getChildren().add(flow);
 
+        updateGridViewItems();
         updateItemCount();
 
         // Register listeners

